infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -417,6 +407,27 @@ private LogReplicationSession constructSession(String sourceClusterId, String si
                 .build();
     }
 
+    private void generateNameForNewlyAddedSession() {
+        for(LogReplicationSession session : newSessionsDiscovered) {
+            String nameForSession = SESSION_NAME_PREFIX + getSessionHash(session);
+            replicationContext.addSessionNameForLogging(session, nameForSession);
+        }
+    }
+
+
+    /**
+     * Implement own hashcode to ensure that hash for a session is always the same (on both source and sink), even on restart.
+     */
+    private int getSessionHash(LogReplicationSession session) {
```

Comment:
hashCode() can return different values across restarts and applications.  So we can run into the same problem with a custom implementation.

Reply:
No, the hashcode should not differ with the custom implementation..since `hashcode()` is either on string or enum.
I have verified that the hashcode remains that same on restarts.

Reply:
Looking at the hashcode implementation for strings, it will not vary.

For enums, it is calculated using Object's hashcode method which uses the runtime address of the object.  This can vary across runs.
https://stackoverflow.com/questions/13581409/how-hashcodes-for-enums-are-calculated-in-java-and-combining-enum-hashcodes-for

https://stackoverflow.com/questions/1516843/java-object-hashcode-result-constant-across-all-jvms-systems

We can use getValue() to get the string form of the model and use hashcode on it to address this.

Reply:
ohk, I thought enums would use the index or string value and not the memory location.
Changed to use the string value.